File: service-invoice/src/main/java/com/acme/services/invoice/java/InvoiceServiceImpl.java
package com.acme.services.invoice.java;

import com.acme.contracts.services.invoice.Invoice;
import com.acme.contracts.services.invoice.InvoiceService;
import com.acme.common.AbstractService;
import com.acme.contracts.workflows.invoicing.User;

import java.time.LocalDateTime;

public class InvoiceServiceImpl extends AbstractService implements InvoiceService {

    @Override
    public Invoice create(User user, double amount, LocalDateTime start, LocalDateTime end) {
        log(String.format("Creating new Invoice for user (%s), amount (%s), start (%s), end (%s)", user.getId(), amount, start, end));
        return new Invoice(user.getId(), amount, start, end);
    }
}

File: service-invoice/src/main/kotlin/com/acme/services/invoice/kotlin/InvoiceServiceImpl.kt
package com.acme.services.invoice.kotlin

import com.acme.contracts.services.invoice.Invoice
import com.acme.contracts.services.invoice.InvoiceService
import com.acme.common.AbstractService
import com.acme.contracts.workflows.invoicing.User
import java.time.LocalDateTime

class InvoiceServiceImpl: AbstractService(), InvoiceService {
    override fun create(user: User, amount: Double, start: LocalDateTime?, end: LocalDateTime?): Invoice {
        log("Creating new Invoice for user (${user.id}), amount ($amount), start ($start), end ($end))")
        return Invoice(user.id, amount, start, end)
    }
}